controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +842,100 @@ var _ = Describe("reconcileRateLimit", func() {
 		Expect(reconcileRateLimit(hetznerCluster, testEnv.RateLimitWaitTime)).To(BeFalse())
 	})
 })
+
+func TestSetControlPlaneEndpoint(t *testing.T) {
+	t.Run("function should return false if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is not enabled and hetznerCluster.Spec.ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: nil}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
+			t.Fatalf("Wrong value for hetznerCluster.Spec.ControlPlaneEndpoint. Got: %s, Want: <nil>", hetznerCluster.Spec.ControlPlaneEndpoint)
+		}
+		assert.Equal(t, false, got)
+	})
+	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is false but hetznerCluster.Spec.ControlPlaneEndpoint is not nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: &clusterv1.APIEndpoint{Host: "xyz"}}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled != false){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint == nil){
+			t.Fatalf("ControlPlaneEndpoint must not be nil")
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint.Host != "xyz"){
+			t.Fatalf("Wrong input for host. Got: %s, Want: 'xyz'", hetznerCluster.Spec.ControlPlaneEndpoint.Host)
+		}
+		assert.Equal(t, true, got)
+	})
+	t.Run("function should return false if ControlPlaneLoadBalancer is enabled and hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 is '<nil>'", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: true}}, Status: infrav1.HetznerClusterStatus{ControlPlaneLoadBalancer: &infrav1.LoadBalancerStatus{IPv4: "<nil>"}}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled != true){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: true", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4 != "<nil>"){
+			t.Fatalf("Wrong value for Status.ControlPlaneLoadBalancer.IPv4. Got: %s, Want: '<nil>'", hetznerCluster.Status.ControlPlaneLoadBalancer.IPv4)
+		}
+		assert.Equal(t, false, got)
```

Comment:
you tested now that there are no changes in ControlPlaneLoadBalancer which is kinda obvious in setControlPlaneEndpoint function, but you didn't check that no changes have been made to the ControlPlaneEndpoint! 

In this test case you would check that it is still nil. So you should have another test case where you have host or port set and check that both are still the same as well.

Reply:
About the second test that you have recommended here, this one is also done below where I check the changes made to ControlPlaneEndpoint (when IPv4 and Port are not nil)

Reply:
ok. then you can maybe remove it